### PR TITLE
[Merged by Bors] - fix: classification utilities (PL-844)

### PR DESCRIPTION
**Pull request by DecathectZero**
A bunch of minor utility changes, mostly defining types and interfaces.

Still need to full review and validate the logic.

One small call out, I didn't add in the new typings for `isHybridLLMStrategy` since it would break things

### CODE REVIEW

**Comment by DecathectZero** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/voiceflow/general-runtime/744?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#744** <a href="https://app.graphite.dev/github/pr/voiceflow/general-runtime/744?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#742** <a href="https://app.graphite.dev/github/pr/voiceflow/general-runtime/742?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#743** <a href="https://app.graphite.dev/github/pr/voiceflow/general-runtime/743?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @DecathectZero and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by theProf — APPROVED** [review]

File: lib/services/classification/interfaces/nlu.interface.ts [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+}
+
+export interface PredictError {
+  message: string;
+}
+
+export interface ClassificationResult {
+  result: 'llm' | 'nlu' | 'nlc';
+  utterance: string;
+  nlc: Partial<{
```
Comment (trs) [inline lib/services/classification/interfaces/nlu.interface.ts:26]: Might help readability to split these sub types out.

**Review by trs — APPROVED** [review]

File: lib/services/classification/interfaces/nlu.interface.ts (hunk 1 above)
Comment (DecathectZero) [reply lib/services/classification/interfaces/nlu.interface.ts:26]: cc @theProf

**Pull request closed without merging**